### ISSUE TITLE
Seeking clarity for key term 'underlying platform software'

### ISSUE THREAD

**Issue by matatk** [w3c/wcag2ict#465]
Seeking clarity for key term 'underlying platform software'

Throughout most of the document, we understand the following uses of two key terms:

* 'Web content' is generally replaced by either 'content' or 'software' - i.e. instead of a web page, we may be considering an app running on a system.

* 'User agent' is generally replaced by 'platform software' - which is defined as 'software that runs on an underlying software or hardware layer and that provides a set of software services to other software components'. To us, this reads as 'the OS, or libraries installed on the system'.

However, in some SCs (and one glossary entry) the following terms are used instead of the above two:

* 'Underlying platform software' seems to be used to mean what we understood 'platform software' to mean above (i.e. the OS, or libraries installed on the system).

* 'Platform software' seems to be used to mean 'an app (running on an OS instead of within a user agent)'

This relates to the following SCs, and a glossary definition:

* 2.5.1 (non-web software branch)

* 2.5.2 (non-web software branch)

* 2.5.7 (non-web software branch)

* 3.3.8

* Glossary entry 'keyboard interface'

What are we asking for?

* If we understand the definitions of the terms correctly, then it seems like there could be an inconsistency with some SCs, as they use the phrase 'underlying platform software'.

* However, if we don't understand correctly, and there is a distinction between 'platform software' and 'underlying platform software', would it be possible to clarify the use of 'underlying platform software' - perhaps adding it to the key terms? 

*[This issue comes from APA WG](https://lists.w3.org/Archives/Public/public-apa-admin/2024Aug/0006.html) - thanks to @newtonsgroove for this issue*

**Comment by GreggVan** [comment]
yes   these should all use the same term == as they mean the same thing.  This should be sent to the AG working group as a candidate for an ERRATA for the WCAG 2.x  that editorially changes these all to use the same words -- for the same concept to avoid exactly the problem that you are pointing out. It is very common in standards to assume that if you do not use the same words to describe the same concept, you did it for a reason, and that you meant the two things to mean something different. That is not the case here so it is a mistake and it should be corrected with an errata.

**Comment by GreggVan** [comment]
Further, after we check with the AG working group, we should make a note in our document that in fact, they should be interpreted as all meaning the same thing and that that was confirmed with the conversation with the AE working group, which is planning to issue an errata.     

(unless of course, somebody in the AG says that they were intended to be separate concepts, and that they are adding a separate definition for the separate concept so that people know what the two different concepts are, and how they differ.) I don't see that happening though. But we should do things in the proper order.

**Comment by maryjom** [comment]
Thank you for taking the time to review the WCAG2ICT public draft. Our task force will review your comment(s) and develop a response that we hope will address your concern(s).

**Comment by maryjom** [comment]
@matatk FYI, the Task Force had already been changing the WCAG2ICT notes you mentioned in your review for SCs in the editor's draft. We are wondering if these changes help to address the confusion expressed in the issue. See:
- [2.5.1 Pointer Gestures](https://w3c.github.io/wcag2ict/#pointer-gestures) (non-web software branch) (Notes 3 and 4)
- [2.5.2 Pointer Cancellation](https://w3c.github.io/wcag2ict/#applying-sc-2-5-2-pointer-cancellation-to-non-web-documents-and-software) (non-web software branch) (Notes 5 and 6)
- [2.5.7 Dragging Movements](https://w3c.github.io/wcag2ict/#applying-sc-2-5-7-dragging-movements-to-non-web-documents-and-software) (non-web software branch) (Notes 3 and 4)

Our intent was not to define another term "underlying platform software" but use "underlying" to indicate that there are software layers - e.g. an application sits on top of some platform software, and sometimes there are layers of platform software (e.g. user agent is platform software that is a layer above the operating system). Just like a web application is not responsible for pointer actions required to operate the user agent or AT, a software application isn't responsible for actions required to operate the platform software it resides on.

What might have added to the confusion is the text styling for any replacement words and phrases in WCAG2ICT. I admit isn't ideal, but it's W3C's styling where the document building script turns `<INS>` markdown into green text with a dotted underline. When there's a definition link inside that inserted text, the link text gets a solid medium gray underline treatment. Visually, they're a bit difficult to discern. We haven't figured out how to define our own CSS to style the inserted text so we can styled it more similar to the 2013 WCAG2ICT Note - that doesn't use a dotted underline at all. 

As for [SC 3.3.8 Accessible Authentication (Minimum)] Note 3 the TF discussed this in today's 16 August working session and we were thinking the use of "underlying platform software" as the replacement phrase was important. Again...simply to indicate the layers of software.

We are OK with removing "Underlying" from Note 1 in the Glossary entry 'keyboard interface', as that word isn't essential to what the note says - it's clear without it. 


Let me know if this isn't satisfactory and we can discuss further.

**Comment by maryjom** [comment]
@matatk @newtonsgroove Thank you for your review and comment. 

In the editor’s draft, the Task Force has already made some changes to the non-web software notes in SCs 2.5.1, 2.5.2, and 2.5.7 due to [Issue 414](https://github.com/w3c/wcag2ict/issues/414). We are wondering if the changes we have already made also address the concern expressed in this issue. See:
- [2.5.1 Pointer Gestures](https://w3c.github.io/wcag2ict/#pointer-gestures) (non-web software branch) (Notes 3 and 4)
- [2.5.2 Pointer Cancellation](https://w3c.github.io/wcag2ict/#applying-sc-2-5-2-pointer-cancellation-to-non-web-documents-and-software) (non-web software branch) (Notes 5 and 6)
- [2.5.7 Dragging Movements](https://w3c.github.io/wcag2ict/#applying-sc-2-5-7-dragging-movements-to-non-web-documents-and-software) (non-web software branch) (Notes 3 and 4)

The Task Force also analyzed and discussed other uses of ‘underlying platform software’ and feel that there is no need for a key term ‘underlying’ or ‘underlying platform software’  (instead of just “Platform software” by itself) for the following reasons: 
- The visual formatting of word replacements (a W3C CSS document styling for INS markdown) may have made it visually appear that the phrase “underlying platform software” is a key term, but it is not. Unfortunately the W3C markup for inserted text (a dotted green underline) can be hard to discern from the medium gray underline used for the link to key terms (so insertion markup looks like key term markup) 
- We are using “underlying” the Mirriam-Webster dictionary meaning which is “lying beneath or below”. Using “underlying” is simply to indicate that there are software layers with the application software being in a layer above the platform layer. There can even be multiple layers of platform software.
- We also think that the changes made for Issue 414 clarifies the related Notes 3 & 4 in SCs 2.5.1 and 2.5.7, and Notes 5 & 6 in SC 2.5.2 by moving “user agent” to the latter note regarding platform software. “User agent” is given as an example in the definition of “platform software”, as well as a statement that there may be multiple layers of platform software. We think these changes brought the notes into alignment with the definition.

For SC 3.3.8 Accessible Authentication, we don’t think any change is needed, as the verbiage is consistent with the 2.5.x SC mentioned above.

For SC 2.1.1 Keyboard and the glossary term “keyboard interface,” the task force agreed that a change to remove “underlying” is reasonable. These instances are both clear without that word.

You can read the changed text in-context in the document using the following links:
- [Applying SC 2.1.1 Keyboard to Non-Web Documents and Software](https://w3c.github.io/wcag2ict/#applying-sc-2-1-1-keyboard-to-non-web-documents-and-software) (See Note 1.)
- [Applying “keyboard interface” to Non-Web Documents and Software](https://w3c.github.io/wcag2ict/#applying-keyboard-interface-to-non-web-documents-and-software) (See Note 1.)

Please respond to let us know that this sufficiently addresses your comment.

**Comment by matatk** [comment]
Thank you @maryjom and the whole TF; your changes address our concerns, so I'm closing this on behalf of APA WG.

Sorry I didn't notice #414 (I did do a bit of searching before filing, but missed it; oops).

*[minutes of the meeting in which we discussed this issue](https://www.w3.org/2024/09/11-apa-minutes#t06)*
